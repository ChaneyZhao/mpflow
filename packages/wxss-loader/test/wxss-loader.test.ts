/* eslint-env jest */
import compile from './helpers/compile'

describe('wxss-loader', () => {
  test('should work', async () => {
    const result = await compile('basic', {})
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('should work with esModule', async () => {
    const result = await compile('basic', { esModule: true })
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('should work with sourceMap', async () => {
    const result1 = await compile('basic', {}, true)
    expect(result1.exports).toMatchSnapshot('exports')
    expect(result1.warnings).toMatchSnapshot('warnings')
    expect(result1.errors).toMatchSnapshot('errors')

    const result2 = await compile('basic', { sourceMap: true })
    expect(result2.exports).toEqual(result1.exports)
    expect(result2.warnings).toEqual(result1.warnings)
    expect(result2.errors).toEqual(result1.errors)
  })

  test('should work with less', async () => {
    const result = await compile('less', {})
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })
})
