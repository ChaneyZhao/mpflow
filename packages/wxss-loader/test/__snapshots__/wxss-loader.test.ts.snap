// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wxss-loader should work with esModule: errors 1`] = `Array []`;

exports[`wxss-loader should work with esModule: exports 1`] = `
Object {
  "default": WxssModule {
    "exports": Array [
      Array [
        "./c.wxss",
        ".c {
  color: green;
}
",
        undefined,
      ],
      Array [
        "./b.wxss",
        ".b {
  color: blue;
  background: url(\\"image.09a1a111.png\\");
}
",
        undefined,
      ],
      Array [
        "./a.wxss",
        ".a {
  color: red;
}
",
        undefined,
      ],
    ],
    "imported": Set {
      "./c.wxss",
      "./b.wxss",
      "./a.wxss",
    },
  },
}
`;

exports[`wxss-loader should work with esModule: warnings 1`] = `Array []`;

exports[`wxss-loader should work with less: errors 1`] = `Array []`;

exports[`wxss-loader should work with less: exports 1`] = `
Object {
  "default": WxssModule {
    "exports": Array [
      Array [
        "./a.less",
        ".c {
  color: green;
}
.b {
  color: blue;
}
.a {
  color: red;
}
",
        undefined,
      ],
    ],
    "imported": Set {
      "./a.less",
    },
  },
}
`;

exports[`wxss-loader should work with less: warnings 1`] = `Array []`;

exports[`wxss-loader should work with sourceMap: errors 1`] = `Array []`;

exports[`wxss-loader should work with sourceMap: exports 1`] = `
Object {
  "default": WxssModule {
    "exports": Array [
      Array [
        "./c.wxss",
        ".c {
  color: green;
}
",
        Object {
          "file": "c.wxss",
          "mappings": "AAAA;EACE,YAAY;AACd",
          "names": Array [],
          "sources": Array [
            "c.wxss",
          ],
          "sourcesContent": Array [
            ".c {
  color: green;
}
",
          ],
          "version": 3,
        },
      ],
      Array [
        "./b.wxss",
        ".b {
  color: blue;
  background: url(\\"image.09a1a111.png\\");
}
",
        Object {
          "file": "b.wxss",
          "mappings": "AAEA;EACE,WAAW;EACX,mDAA8B;AAChC",
          "names": Array [],
          "sources": Array [
            "b.wxss",
          ],
          "sourcesContent": Array [
            "@import url('./c.wxss');

.b {
  color: blue;
  background: url('./image.png');
}
",
          ],
          "version": 3,
        },
      ],
      Array [
        "./a.wxss",
        ".a {
  color: red;
}
",
        Object {
          "file": "a.wxss",
          "mappings": "AAEA;EACE,UAAU;AACZ",
          "names": Array [],
          "sources": Array [
            "a.wxss",
          ],
          "sourcesContent": Array [
            "@import \\"b.wxss\\";

.a {
  color: red;
}
",
          ],
          "version": 3,
        },
      ],
    ],
    "imported": Set {
      "./c.wxss",
      "./b.wxss",
      "./a.wxss",
    },
  },
}
`;

exports[`wxss-loader should work with sourceMap: warnings 1`] = `Array []`;

exports[`wxss-loader should work: errors 1`] = `Array []`;

exports[`wxss-loader should work: exports 1`] = `
Object {
  "default": WxssModule {
    "exports": Array [
      Array [
        "./c.wxss",
        ".c {
  color: green;
}
",
        undefined,
      ],
      Array [
        "./b.wxss",
        ".b {
  color: blue;
  background: url(\\"image.09a1a111.png\\");
}
",
        undefined,
      ],
      Array [
        "./a.wxss",
        ".a {
  color: red;
}
",
        undefined,
      ],
    ],
    "imported": Set {
      "./c.wxss",
      "./b.wxss",
      "./a.wxss",
    },
  },
}
`;

exports[`wxss-loader should work: warnings 1`] = `Array []`;
